packages/bot-web-ui/src/pages/dashboard/announcements/announcements.tsx

```diff
@@ -0,0 +1,172 @@
+import { useHistory } from 'react-router-dom';
+import clsx from 'clsx';
+import { localize } from '@deriv/translations';
+import AnnouncementDialog from './announcement-dialog';
+import { BOT_ANNOUNCEMENTS_LIST, TAnnouncement, TNotifications } from './config';
+import './announcements.scss';
+import { MessageAnnounce, TitleAnnounce } from './announcement-components';
+import { getButtonAction } from './utils/accumulator-helper-functions';
+
+type TAnnouncements = {
+    is_mobile?: boolean;
+    handleTabChange: (item: number) => void;
+};
+
+const Announcements = ({ is_mobile, handleTabChange }: TAnnouncements) => {
+    const [is_announce_dialog_open, setIsAnnounceDialogOpen] = useState(false);
+    const [is_open_announce_list, setIsOpenAnnounceList] = React.useState(false);
+    const [selected_announcement, setSelectedAnnouncement] = React.useState<TAnnouncement | null>(null);
+    const [stored_notifications, setStoredNotifications] = React.useState({} as Record<string, boolean>);
+    const history = useHistory();
+    const [notifications, setNotifications] = useState([] as TNotifications[]);
+    const action_button_class_name = 'announcements__label';
+
+    const storeDataInLocalStorage = (temp_data: Record<string, boolean>) => {
+        localStorage?.setItem('bot-announcements', JSON.stringify(temp_data));
+    };
+
+    const modalButtonAction = (announce_id: string, announcement: TAnnouncement) => () => {
+        setSelectedAnnouncement(announcement);
+        setIsAnnounceDialogOpen(true);
+        setIsOpenAnnounceList(prev => !prev);
+
+        let data: Record<string, boolean> | null = null;
+        data = JSON.parse(localStorage.getItem('bot-announcements') ?? '{}');
+
+        storeDataInLocalStorage({ ...data, [announce_id]: false });
+        const temp_notifications = updateNotifications();
+        setStoredNotifications(temp_notifications);
+    };
+
+    const handleRedirect = (url: string) => {
+        if (history) {
+            history.push(url);
+        }
+    };
+
+    const updateNotifications = () => {
+        let data: Record<string, boolean> | null = null;
+        data = JSON.parse(localStorage.getItem('bot-announcements') ?? '{}');
+        const tmp_notifications: TNotifications[] = [];
+        const temp_localstorage_data: Record<string, boolean> | null = {};
+        BOT_ANNOUNCEMENTS_LIST.map(item => {
+            let is_not_read = true;
+            if (data && Object.hasOwn(data, item.id)) {
+                is_not_read = data[item.id];
+            }
+            tmp_notifications.push({
+                key: item.id,
+                icon: <item.icon announce={is_not_read} />,
+                title: <TitleAnnounce title={item.title} announce={is_not_read} />,
+                message: <MessageAnnounce message={item.message} date={item.date} announce={is_not_read} />,
+                buttonAction: getButtonAction(item, modalButtonAction, handleRedirect),
```

Comment:
1) The name of the `getButtonAction` method really doesn't capture its essence, because the method doesn't just ‘get’ an action, it executes it. 
Will be it fine to use `performButtonAction` name instead?

2)
We use the local state, update it, and according to it, we rerender the look of announcements. We need it here.
Yeah, we mostly keep getButtonAction() in the utils file to keep the component readable with less logic because the main thing that should do the component is just to have what we are going to render. I don't see anything wrong with that, as long as that method can easily exist elsewhere and is easier to maintain. If I am wrong please change my mind or correct me. Maybe there are some other options😌🧐

Reply:
@wojciech-deriv We make an iteration to define or provide our announcements with functions that will be called once the user clicks on one of them. The `performButtonAction` method will be called so many times how many announcements we have. This is expected. In general, this action happens only once.

Reply:
no, when we use the method `performButtonAction` we use `switch () case` that use only 1 option

Reply:
I checked, in this case it won't open the modal window on click